gax/src/main/java/com/google/api/gax/retrying/NoopRetryingContext.java

```diff
@@ -51,4 +53,14 @@ public static RetryingContext create() {
   public ApiTracer getTracer() {
     return NoopApiTracer.getInstance();
   }
+
+  @Override
+  public RetrySettings getRetrySettings() {
+    return null;
```

Comment:
perhaps a retry settings that does nothing instead of null

Reply:
`RetrySettings` don't really have an empty default value. In addition, the contract for the existing interface `RetryingContext` is that any implementation should return `null` if the default settings should not be overridden.